src/Chapter2.hs

```diff
@@ -781,13 +803,16 @@ Let's now try to eta-reduce some of the functions and ensure that we
 mastered the skill of eta-reducing.
 -}
 divideTenBy :: Int -> Int
-divideTenBy x = div 10 x
+divideTenBy = div 10
 
 -- TODO: type ;)
-listElementsLessThan x l = filter (< x) l
+listElementsLessThan :: Int -> [Int] -> [Int]
+listElementsLessThan x = filter (< x)
 
 -- Can you eta-reduce this one???
-pairMul xs ys = zipWith (*) xs ys
+pairMul :: Num c => [c] -> [c] -> [c]
+--pairMul xs ys = zipWith (*) xs ys
+pairMul = zipWith (*)
```

Comment:
Eta-reduction award! 🏆 

I totally understand that understanding and working with eta-reduction can be challenging at first.

I can give some more explanation on the topic.

When eta-reducing, you need to remove arguments from both sides of the `=` sign. So the correct solution is:

```haskell
divideTenBy = div 10
```

See how it compares with the original code:

```diff
-divideTenBy x = div 10 x
+divideTenBy = div 10
```

The motivation behind eta-reduction if that if you have two functions `foo` and `bar` of the same type, and they return the same results for any arguments (so `foo x` always equal to `bar x` for any `x`). In such situations, we say that functions `foo` and `bar` are equal, and if we want to define `foo` in terms of `bar`, we can write `foo = bar` instead of a longer `foo x = bar x` (which means that `foo` returns the same result as `bar` for any `x`).

I hope it helps 🙂

Reply:
It does, Thanks a lot!